### PR TITLE
GIT-48 Implement game component

### PR DESCRIPTION
**Pull request by kate20031**

### CODE REVIEW

File: frontend/src/components/Game.js [hunk 1]
```diff
@@ -1,13 +1,198 @@
...
+        },
+        body: JSON.stringify({
+          answer: answer,
+        }),
+      };
+
+      fetch(`${BACKEND_URL}/game/${gameId}`, options)
+        .then((response) => {
+          if (response.ok) {
+            setAnswerSubmitted(true);
```
Comment (lukaszsmolinski) [inline frontend/src/components/Game.js:125]: I don't think we need the `answerSubmitted` flag. I think that after submitting the answer, the player should be automatically redirected to the results.

File: frontend/src/components/Game.js [hunk 2]
```diff
@@ -1,13 +1,198 @@
...
+      <div className="form">
+        <div className="info">
+          Player: {playerName}
+          <br />
+          Current path: {currentPath}
+        </div>
+        <h2 className="directories">Directories:</h2>
+        <button className="backButton" onClick={handleReturn}>
+          Back
+        </button>
```
Comment (lukaszsmolinski) [inline frontend/src/components/Game.js:160]: Can you make this button disabled when the player is already in the root directory?

File: frontend/src/components/Game.js [hunk 3]
```diff
@@ -1,13 +1,198 @@
...
+                Choose
+              </button>
+            </li>
+          ))}
+        </ul>
+        {answer && <div className="chosenDir">Chosen directory: {answer}</div>}
+        <button
+          type="submit"
+          className="buttonSubmit"
+          onClick={() => handleSubmit(answer)}
```
Comment (lukaszsmolinski) [inline frontend/src/components/Game.js:189]: Can you make this button disabled when the player hasn't chosen any directory yet?

File: frontend/src/components/Game.js [hunk 4]
```diff
@@ -1,13 +1,198 @@
...
 function Game() {
+  const [error, setError] = useState(null);
   const location = useLocation();
-  const state = location.state;
+  const { gameId } = location.state;
+  const [playerName, setPlayerName] = useState("");
+  const [repositoryId, setRepositoryId] = useState("");
+  const [directories, setDirectories] = useState([]);
+  const [pathHistory, setPathHistory] = useState([]);
+  const [currentPath, setCurrentPath] = useState("gitguesser");
```
Comment (lukaszsmolinski) [inline frontend/src/components/Game.js:14]: The path shouldn't begin with `gitguesser` in every repository.

**Review by Virtover — CHANGES_REQUESTED** [review]
1) Add the name of the file whose location we want to guess
2) Redirect to Results (with sending gameId) when answer is submitted

**Review by Virtover — CHANGES_REQUESTED** [review]
image
in this case (and others when I click directory without subdirectories), location "/frontend/src/components" is not chosen when I click submit

**Review by Virtover — CHANGES_REQUESTED** [review]
-> Answer should be full path, but only folder name is send as an answer. This should be changed, otherwise game will not work.

-> Choose button is in the center when there is no back button, and flies to the right when there is back button. I think it would be good to make it staying in one place.

**Review by Virtover — APPROVED** [review]

**Review by lukaszsmolinski — APPROVED** [review]

**Pull request merged**